### ISSUE TITLE
Issue with application.properties auto completion

### ISSUE THREAD

**Issue by wedamartinez** [redhat-developer/intellij-quarkus#49]
Since I have installed the quarkus IntelliJ plugin I'm not able to edit the application.properties, intellij get frozen after writing some lines.

Also if do copy and paste a part of the property it gets duplicated when I try to finish the property tag.

My IntelliJ IDEA 2019.3.2 Preview (Ultimate Edition)
macOS 10.15.2
Thanks!

**Comment by jeffmaury** [comment]
Which version of the plugin are you using ?
Can you describe the steps you follow when editing your application.properties ? Or give us a sample ?

**Comment by wedamartinez** [comment]
Yes of course, Quarkus Tools version 3.2

**Comment by thiadmer** [comment]
Also getting freezes from time to time. Often when quickly copy-pasting in the same application.properties IntelliJ hangs completely on 'Computing property definition'. I can send you my props file if needed.
macOS 11.1 bigsur, Idea 2020.3, Quarkus Tools 1.2.0.71

**Comment by tlvlp** [comment]
I'm experiencing the same problem right now with trying to set quarkus datasoruces. It is doing the same every time, even after a reboot. 
Intellij (2020.3 ultimate) Completely freezes and does not seem to be able to recover. Running on AMD Ryzen7 4800H with 8 cores and 32 GB RAM, so I don't think that it would be a resource-related problem.

**Comment by jeffmaury** [comment]
@thiadmer is there a way I can get your properties file ?
@tlvlp can you describe your scenario as I can't get to reproduce the problem.
I you can the thread dump when the problem happens, it would be very helpful

**Comment by tlvlp** [comment]
@jeffmaury It happens often when I'm trying to modify quarkus datasource properties for a project with two separate datasources. Specifically copying a named datasource's name to one or multiple lines at the same time (inserting it in the middle of the property names) eg.:
quarkus.datasource.**INSERTED_NAME**.jdbc.url

Curiously it seems to be happening when I:
1. Change the name of the datasource in multiple lines at once with the ctrl + shift selection
2. Try to copy over the datasource name form the `quarkus.datasource.` to the `quarkus.hibernate-orm.` properties eg. The freeze occurs at this step.
```
quarkus.datasource.**NEW_DS_NAME**.db-kind=mysql    <----- 1. modified together 
quarkus.datasource.**NEW_DS_NAME**.jdbc.url=jdbc:mysql://${HOST}    <----- 1. modified together 
quarkus.datasource.**NEW_DS_NAME**.username=${USER}    <----- 1. modified together     
quarkus.datasource.**NEW_DS_NAME**.password=${PASS}    <----- 1. modified together 
quarkus.datasource.**NEW_DS_NAME**.jdbc.initial-size=1    <----- 1. modified together 
quarkus.datasource.**NEW_DS_NAME**.jdbc.min-size=1    <----- 1. modified together 
quarkus.datasource.**NEW_DS_NAME**.jdbc.max-size=5    <----- 1. modified together 
quarkus.hibernate-orm.**OLD_DS_NAME**.datasource=**OLD_DS_NAME** <--------- 2. Ctrl+v / Ctrl+v the **NEW_DS_NAME** here
quarkus.hibernate-orm.**OLD_DS_NAME**.packages=com.mypackage
```
 
I have attached the auto thread dumps from jan 28 (when it originally happened) and from now when I reproduced it with the steps above. Note that today's logs seem a bit shorter with less information as I killed the process sooner.

[intellij_hangs_on_app_props_edit.zip](https://github.com/redhat-developer/intellij-quarkus/files/5943666/intellij_hangs_on_app_props_edit.zip)

**Comment by jeffmaury** [comment]
@tlvlp which version of Quarkus Tools are you on ? 
Can you describe the list of actions because I can't reproduce and don't understand the ctrl + shift (which is key as I remember I fixed something related to Goto action which I can see being blocked in your various stack traces)

**Comment by tlvlp** [comment]
@jeffmaury  my Quarkus Tools version is: 1.3.0.103

Sorry, I was wrong about the **ctrl + shift** part. I was using **left alt + mouse drag** for the [rectangular text range selection](https://www.jetbrains.com/help/rider/Multicursor.html) feature of Intellij. The main impact of this could be that the middle section of multiple quarkus.datasource properties are changed at the same time.
The freeze has occurred in the next step, when right after the previous change I have copied the `NEW_DS_NAME` to the quarkus.hibernate-orm.`OLD_DS_NAME`. Not sure if the sequence matters, maybe it has something to do with the quarkus.hibernate-orm settings.

**Comment by jeffmaury** [comment]
@tlvlp sorry to bother you again but I can't reproduce. Here what I did:

1. Initial config file is:

```
quarkus.datasource.OLD_DS_NAME.db-kind=mysql 
quarkus.datasource.OLD_DS_NAME.password=${PASS} 
quarkus.datasource.OLD_DS_NAME.jdbc.initial-size=1 
quarkus.datasource.OLD_DS_NAME.jdbc.min-size=1 
quarkus.datasource.OLD_DS_NAME.jdbc.max-size=5 
quarkus.datasource.OLD_DS_NAME.username=${USER} 
quarkus.datasource.OLD_DS_NAME.jdbc.url=jdbc:mysql://${HOST} 
quarkus.hibernate-orm.OLD_DS_NAME.datasource=OLD_DS_NAME
quarkus.hibernate-orm.OLD_DS_NAME.packages=com.mypackage
```

2. Using Left Alt + Mouse Drag select `OLD_DS_NAME` in the `quarkus.datasource` section and change the name to `NEW_DS_NAME`

3. Copy NEW_DS_NAME to the clipboard (double click on one of the `NEW_DS_NAME` and `Ctrl + C`)
4. Select `OLD_DS_NAME` and paste (`Ctrl + V`)

**Comment by tlvlp** [comment]
@jeffmaury  indeed your steps were correct.  I have double checked and found that the issue cannot be reproduced in an empty project so I have tried to build a minimal reproducer adding features one by one from the project where the problem is steadily reproducable.

Steps:
1. Clone https://github.com/tlvlp/quarkus_app_prop_freeze_example
2. Run maven build
3. Start project in quarkus dev mode (after a few attempts I managed to reproduce without starting it as well).
4. Ignore the database connection errors :) the app will still keep running. The original project had proper DB connections and the freeze still occurs so this should not be a factor.
5. Using Left Alt + Mouse Drag select `OLD_DS_NAME` in the `quarkus.datasource` section and change the name to `NEW_DS_NAME`.
6. Copy NEW_DS_NAME to the clipboard (double click on one of the NEW_DS_NAME and Ctrl + C)
7. Select OLD_DS_NAME and paste (Ctrl + V) <--- In some cases I had to repeat this step 2-3 times, in others it instantly hangs.

**Notes:** 
- I just noticed that the bottom-right status indicator is stuck in **"Computing property definition"**.
- All CPU cores are running around 20% and the java processes consume at max 2 / 32 GB. 
- Nothing responds on any of the open Intellij windows. Not even to SIGTERM, only SIGKILL works.

I have also attached the thread dumps from the freezes produced with the new test project:
[threadDumps-freeze-20210208-2.zip](https://github.com/redhat-developer/intellij-quarkus/files/5947173/threadDumps-freeze-20210208-2.zip)

**Comment by shumonsharif** [comment]
I am experiencing the same behavior described here - IntelliJ becomes completely unresponsive while trying to modify properties files, and only SIGKILL works to terminate the IDE.  I am able to reproduce the issue consistently.

I am on Quarkus Tools 1.4.0.121 with IntelliJ Ultimate 2020.3.2.  Thread dumps from both the IntelliJ and the Quarkus Tools processes are attached.

[thread-dumps.zip](https://github.com/redhat-developer/intellij-quarkus/files/6153024/thread-dumps.zip)

**Comment by jeffmaury** [comment]
@angelozerr that's something you might look it as it seems the LSP is in an infinite loop

**Comment by angelozerr** [comment]
@datho7561 I suspect a problem with property expression. What do you think about that?

**Comment by datho7561** [comment]
Has this been merged into intellij-quarkus?: https://github.com/eclipse/lsp4mp/commit/c3e3023ffcb1432a18291280a30bf72de302be06

**Comment by jeffmaury** [comment]
No intellij-quarkus is based on released versions of LSP4MP and quarkus-ls

**Comment by datho7561** [comment]
Okay, it could be https://github.com/eclipse/lsp4mp/issues/112. I'll see if these steps can reproduce it consistently on the latest released version of intellij-quarkus.

**Comment by shumonsharif** [comment]
I just verified that it is indeed the same tabs issue causing IntelliJ to hang for me.  I searched for any lines with tab characters in my properties file using:

`grep $'\t' src/main/resources/application.properties`

Removed the trailing tab from the property in question, and IntelliJ is working fine now.

**Comment by jeffmaury** [comment]
Fixed by 43d2a9adbfa5b2c541943ac593da2acc99769038
